### ISSUE TITLE
ERR_CONNECTION_RESET when new MAS app update is installed

### ISSUE THREAD

**Issue by ghost** [electron/electron#22273]
### Preflight Checklist

* [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
* [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [X] I have searched the issue tracker for an issue that matches the one I want to file, without success.

### Issue Details

* **Electron Version:**
  * 5.0.11 , 7.0.1
* **Operating System:**
  * macOS mas builds
* **Last Known Working Electron version:**
  * N/A

### Expected Behavior

When the app is updated from store it should reach the requested domain without throwing 
any error
### Actual Behavior

The app is updated in the background from AppStore.
When the app is opened again, 
`did-fail-load` event is fired with below error
```
errorCode : -101
errorDescription : ERR_CONNECTION_RESET
```
Error code is thrown even after restarting the app multiple times.
Fixed only after restarting the machine

### To Reproduce

Let the app get updated in the background from AppStore when the machine is locked/idle.
when the App loads remote content in the browser window, did-fail-load event is fired

### Additional Information

On checking, the request to the remote site did not reach the server. i suspect it got failed even before it was made.
This [Issue](https://www.bountysource.com/issues/52874892-cannot-catch-errors-like-net-err_network_changed-net-err_connection_reset) also looks similar but deals with autoUpdater module. According to the electron docs autoUpdater module is [disabled ](https://www.electronjs.org/docs/tutorial/mac-app-store-submission-guide#limitations-of-mas-build) and not sure if this affects mas build.

**Comment by soundar78** [comment]
Update :
This even happens when the app is getting installed for the first time and mac changes to sleep mode during installation.

**Comment by electron-triage** [comment]
The Electron version reported on this issue is no longer supported. See our [supported versions documentation](https://www.electronjs.org/docs/tutorial/support#supported-versions).

If this is still reproducible on a supported version, please open a new issue with any other new information that a maintainer should know.

Thank you for taking the time to report this issue and helping to make Electron better! Your help is appreciated.
